### DISCUSSION TITLE
GitHub Actions Feature Roadmap

### DISCUSSION THREAD

**Discussion by dallinwright** [community/community#27218] [Actions]
<p>There are certain features and requests I routinely see for GitHub actions and after searching the forum can find no apparent solution. Is there a public location with a feature roadmap of what new features are in progress and approximate timelines for them? For example, organization and team level secrets and their management?</p>

**Comment by Yanjingzhu** [comment]
<p>Currently , Github Actions doesn’t provide a Feature Roapmap to public . Secrets API will be published at the end of this month.</p>
<p>In addition, there is the github changelog: <a href="https://github.blog/changelog/" rel="nofollow noopener">https://github.blog/changelog/</a> All the updates of Github Actions will be shown there. </p>

**Comment by dexterlakin** [comment]
<p>I’d really like to have some visibility of what’s on the GA roadmap.</p>
<p>I’ve just moved to a new organisation and we’re currently in the early stages of migrating from a self-hosted Jenkins stack to a managed solution.</p>
<p>We’re considering GA amongst other alternatives (Concourse, Travis, etc.). I’d like us to adopt GA as the organisation is already using Github, but currently GA is lagging quite far behind those other options (lack of support for templating being just one example).</p>
<p>It’s hard to make a case for it though, without a clear picture of what features are being worked on.</p>
<p>Here’s a good example of how the Azure Devops team communicates its roadmap:</p>
<p><a href="https://docs.microsoft.com/en-us/azure/devops/release-notes/features-timeline" rel="nofollow noopener">https://docs.microsoft.com/en-us/azure/devops/release-notes/features-timeline</a></p>
<p><a href="https://dev.azure.com/mseng/AzureDevOpsRoadmap" rel="nofollow noopener">https://dev.azure.com/mseng/AzureDevOpsRoadmap</a></p>
<p><a href="https://devblogs.microsoft.com/devops/top-stories-from-the-microsoft-devops-community-2020-03-27/" rel="nofollow noopener">https://devblogs.microsoft.com/devops/top-stories-from-the-microsoft-devops-community-2020-03-27/</a></p>
<p>Could we have something similar for GA?</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/dexterlakin">@dexterlakin</a> Thank you for pointing this out.  For the request for GA roadmap , could you please share your idea in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">Feedback form for GitHub Actions</a> ?</p>

**Comment by dexterlakin** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> I’ve raised the feedback now.</p>

**Comment by dexterlakin** [comment, marked as answer]
<p>For anyone else that missed it you can find the public roadmap here now:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/github/roadmap/projects/1" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/27975947/181102272-46d01ca3-7924-4b8b-85ee-c23fac6ffa8b.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfe24cb1d118bbfd5efedc3b014833395ad10d0c_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfe24cb1d118bbfd5efedc3b014833395ad10d0c_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/f/dfe24cb1d118bbfd5efedc3b014833395ad10d0c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfe24cb1d118bbfd5efedc3b014833395ad10d0c_2_10x10.png"></div>

<h3><a href="https://github.com/github/roadmap/projects/1" target="_blank" rel="noopener nofollow ugc">github/roadmap</a></h3>

<p>GitHub public roadmap. Contribute to github/roadmap development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
